src/Menu/MenuItem.js

```diff
@@ -53,13 +63,29 @@ class MenuItem extends React.PureComponent {
         }}
       >
         <input {...{ id, name, value, css: hideVisually(), type: 'hidden' }} />
-        <Icon
-          name="symbol-circle"
-          mr="small"
-          color={baseColor}
-          fontSize="16px"
-        />
-        <Text.span fontSize="size4">{label}</Text.span>
+        <Flex width="100%">
+          <Flex
+            flex="1 0 auto"
+            minWidth="120px"
+            maxWidth="100%"
+            overflow="hidden"
+          >
+            <Icon
+              name="symbol-circle"
+              mr="small"
+              color={baseColor}
+              fontSize="16px"
+            />
+            <Text.p fontSize="size4">{label}</Text.p>
```

Comment:
changed from span to p so that getting the desired flex behavior/overflow/width isn't an annoying thing